packages/path_provider/path_provider_android/test/path_provider_android_test.dart

```diff
@@ -0,0 +1,119 @@
+    final List<MethodCall> log = <MethodCall>[];
+
+    setUp(() async {
+      pathProvider = PathProviderAndroid();
+      TestDefaultBinaryMessengerBinding.instance!.defaultBinaryMessenger
+          .setMockMethodCallHandler(pathProvider.methodChannel,
+              (MethodCall methodCall) async {
+        log.add(methodCall);
+        switch (methodCall.method) {
+          case 'getTemporaryDirectory':
+            return kTemporaryPath;
+          case 'getApplicationSupportDirectory':
+            return kApplicationSupportPath;
+          case 'getApplicationDocumentsDirectory':
+            return kApplicationDocumentsPath;
+          case 'getStorageDirectory':
+            return kStoragePath;
+          case 'getExternalCacheDirectories':
+            return kExternalCachePaths;
+          case 'getExternalStorageDirectories':
+            return kExternalStoragePaths;
+          default:
+            return null;
+        }
+      });
+    });
+
+    tearDown(() {
+      log.clear();
+    });
+
+    test('getTemporaryPath', () async {
+      final String? path = await pathProvider.getTemporaryPath();
+      expect(
+        log,
+        <Matcher>[isMethodCall('getTemporaryDirectory', arguments: null)],
+      );
+      expect(path, kTemporaryPath);
+    });
+
+    test('getApplicationSupportPath', () async {
+      final String? path = await pathProvider.getApplicationSupportPath();
+      expect(
+        log,
+        <Matcher>[
+          isMethodCall('getApplicationSupportDirectory', arguments: null)
+        ],
+      );
+      expect(path, kApplicationSupportPath);
+    });
+
+    test('getApplicationDocumentsPath', () async {
+      final String? path = await pathProvider.getApplicationDocumentsPath();
+      expect(
+        log,
+        <Matcher>[
+          isMethodCall('getApplicationDocumentsDirectory', arguments: null)
+        ],
+      );
+      expect(path, kApplicationDocumentsPath);
+    });
+
+    test('getExternalStoragePath', () async {
+      final String? result = await pathProvider.getExternalStoragePath();
+      expect(
+        log,
+        <Matcher>[isMethodCall('getStorageDirectory', arguments: null)],
+      );
+      expect(result, kStoragePath);
+    });
+
+    test('getExternalCachePaths', () async {
+      final List<String>? result = await pathProvider.getExternalCachePaths();
+      expect(
+        log,
+        <Matcher>[isMethodCall('getExternalCacheDirectories', arguments: null)],
+      );
+      expect(result, kExternalCachePaths);
+    });
+
+    test('getExternalStoragePaths', () async {
+      final List<String>? result = await pathProvider.getExternalStoragePaths();
+      expect(
+        log,
+        <Matcher>[
+          isMethodCall('getExternalStorageDirectories', arguments: null)
+        ],
+      );
+      expect(result, kExternalStoragePaths);
+    });
+
+    test('getLibraryPath throws', () async {
+      expect(pathProvider.getLibraryPath(), throwsA(isUnsupportedError));
```

Comment:
These are failing; you either need to make the functions async, or use `() => pathProvider.foo()` in the `expect`.